### PR TITLE
volume-migration: remove restriction for the hotplug volumes

### PR DESCRIPTION
**Pull request by alicefr**
### What this PR does
Before this PR:
Volume migration wasn't allowed for hotplugged volumes

After this PR:
Enable volume migration for volume hotplug

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
Enable volume migration for hotplugged volumes
```

### CODE REVIEW

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmi-update-admitter.go [hunk 1]
```diff
@@ -177,11 +177,13 @@ func admitStorageUpdate(newVolumes, oldVolumes []v1.Volume, newDisks, oldDisks [
...
 
-func verifyHotplugVolumes(newHotplugVolumeMap, oldHotplugVolumeMap map[string]v1.Volume, newDisks, oldDisks map[string]v1.Disk) *admissionv1.AdmissionResponse {
+func verifyHotplugVolumes(newHotplugVolumeMap, oldHotplugVolumeMap map[string]v1.Volume, newDisks, oldDisks map[string]v1.Disk,
+	migratedVols map[string]bool) *admissionv1.AdmissionResponse {
 	for k, v := range newHotplugVolumeMap {
 		if _, ok := oldHotplugVolumeMap[k]; ok {
+			_, okMigVol := migratedVols[k]
 			// New and old have same volume, ensure they are the same
-			if !equality.Semantic.DeepEqual(v, oldHotplugVolumeMap[k]) {
+			if !equality.Semantic.DeepEqual(v, oldHotplugVolumeMap[k]) && !okMigVol {
```
Comment (mhenriks) [inline pkg/virt-api/webhooks/validating-webhook/admitters/vmi-update-admitter.go:186]: unit test?

File: pkg/virt-controller/watch/volume-migration/volume-migration_test.go [hunk 2]
```diff
@@ -81,11 +81,6 @@ var _ = Describe("Volume Migration", func() {
 			), libvmi.NewVirtualMachine(libvmi.New(
 				libvmi.WithPersistentVolumeClaim("disk0", "vol2"), withFilesystemVolume("disk1", "vol4"),
 			)), fmt.Errorf("invalid volumes to update with migration: filesystems: [disk1]")),
-			Entry("with an invalid hotplugged volume", libvmi.New(
-				libvmi.WithPersistentVolumeClaim("disk0", "vol0"), withFilesystemVolume("disk1", "vol1"),
-			), libvmi.NewVirtualMachine(libvmi.New(
-				libvmi.WithPersistentVolumeClaim("disk0", "vol2"), withHotpluggedVolume("disk1", "vol4"),
-			)), fmt.Errorf("invalid volumes to update with migration: hotplugged: [disk1]")),
 		)
```
Comment (mhenriks) [inline pkg/virt-controller/watch/volume-migration/volume-migration_test.go:84]: Should there be a positive hotplug volume test?

**Review by mhenriks — COMMENTED** [review]
looks good

File: pkg/virt-controller/watch/volume-migration/volume-migration_test.go (hunk 2 above)
Comment (alicefr) [reply pkg/virt-controller/watch/volume-migration/volume-migration_test.go:89]: You are right, I made the entry as a valid option

**Comment by alicefr** [comment]
@mhenriks PTAL

**Review by lyarwood — CHANGES_REQUESTED** [review]
Looks like you need to rebase and run `make generate` again, I couldn't reproduce this against the PR branch FWIW so I guess someone has removed this inbetween on main:

```
$ git rebase origin/main
[..]
$ make generate
[..]
$ git diff
diff --git a/tests/storage/migration.go b/tests/storage/migration.go
index 43ed6c68c5..b4cca7f05c 100644
--- a/tests/storage/migration.go
+++ b/tests/storage/migration.go
@@ -38,6 +38,7 @@ import (
        "k8s.io/apimachinery/pkg/types"
        "k8s.io/apimachinery/pkg/util/rand"
 
+       v1 "kubevirt.io/api/core/v1"
        virtv1 "kubevirt.io/api/core/v1"
        "kubevirt.io/client-go/kubecli"
        cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
```

**Comment by alicefr** [comment]
@mhenriks @lyarwood can you please take a look again to this?

**Review by awels — APPROVED** [review]

**Pull request merged**